lib/archive/archive.go

```diff
@@ -0,0 +1,97 @@
+package archive
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/pierrec/lz4/v4"
+)
+
+func DecompressTarLz4(ctx context.Context, inputFile string, outputDir string) error {
```

Comment:
We do AHA instead of DRY (see go style guide), just put this file in cil/cmd folder for now.

Reply:
Done.